src/main.ts

```diff
@@ -0,0 +1,24 @@
+import * as wallet from "./scripts/connect-wallet";
+import * as ui from "./scripts/ui";
+import * as mint from "./scripts/mint";
+import * as redeem from "./scripts/redeem";
+
+export async function mainModule() {
+  ui.initUiEvents();
+
+  wallet.updateConnectButtonText("");
+
+  await wallet.connectIfAuthorized();
+  await mint.initCollateralList();
```

Comment:
The collateral list is shared between mint and redeem. Making it global is a good idea.